### PR TITLE
lvsetup: Using loop device APIs

### PR DESCRIPTION
**Pull request by narasimhan-v**
Using the loop device APIs to create and delete loop devices.

Signed-off-by: Narasimhan V <sim@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by narasimhan-v** [comment]
https://github.com/avocado-framework/avocado/pull/3966

**Comment by narasimhan-v** [comment]
```
# avocado run lvsetup.py
JOB ID     : 44c7c63c5c8af26ce417b0fd991a79f028e532ae
JOB LOG    : /root/sim/tests/results/job-2020-06-24T17.27-44c7c63/job.log
 (1/3) lvsetup.py:Lvsetup.test: PASS (22.21 s)
 (2/3) lvsetup.py:Lvsetup.test_vg_reactivate: PASS (65.54 s)
 (3/3) lvsetup.py:Lvsetup.test_lv_snapshot: PASS (29.78 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sim/tests/results/job-2020-06-24T17.27-44c7c63/results.html
JOB TIME   : 120.06 s
```
```
# avocado run lvsetup.py -m lvsetup.py.data/lvsetup.yaml 
JOB ID     : 8b0033717a48afef098f42a7d4d884cfa66d52de
JOB LOG    : /root/sim/tests/results/job-2020-06-24T17.30-8b00337/job.log
 (1/9) lvsetup.py:Lvsetup.test;ext4-lv_size-42fc: PASS (16.60 s)
 (2/9) lvsetup.py:Lvsetup.test;xfs-lv_size-74e0: PASS (16.55 s)
 (3/9) lvsetup.py:Lvsetup.test;btrfs-lv_size-060d: CANCEL: btrfs is not supported with RHEL 7.4 onwards (4.44 s)
 (4/9) lvsetup.py:Lvsetup.test_vg_reactivate;ext4-lv_size-42fc: PASS (59.10 s)
 (5/9) lvsetup.py:Lvsetup.test_vg_reactivate;xfs-lv_size-74e0: PASS (59.76 s)
 (6/9) lvsetup.py:Lvsetup.test_vg_reactivate;btrfs-lv_size-060d: CANCEL: btrfs is not supported with RHEL 7.4 onwards (4.27 s)
 (7/9) lvsetup.py:Lvsetup.test_lv_snapshot;ext4-lv_size-42fc: PASS (23.52 s)
 (8/9) lvsetup.py:Lvsetup.test_lv_snapshot;xfs-lv_size-74e0: PASS (20.74 s)
 (9/9) lvsetup.py:Lvsetup.test_lv_snapshot;btrfs-lv_size-060d: CANCEL: btrfs is not supported with RHEL 7.4 onwards (3.84 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 3
JOB HTML   : /root/sim/tests/results/job-2020-06-24T17.30-8b00337/results.html
JOB TIME   : 211.96 s
```

**Comment by narasimhan-v** [comment]
```
17:27:53 DEBUG| PARAMS (key=lv_disks, path=*, default=None) => None
17:27:53 DEBUG| PARAMS (key=vg_name, path=*, default=avocado_vg) => 'avocado_vg'
17:27:53 DEBUG| PARAMS (key=lv_name, path=*, default=avocado_lv) => 'avocado_lv'
17:27:53 DEBUG| PARAMS (key=fs, path=*, default=ext4) => 'ext4'
17:27:53 DEBUG| PARAMS (key=lv_size, path=*, default=0) => '0'
17:27:53 DEBUG| PARAMS (key=lv_snapshot_name, path=*, default=avocado_sn) => 'avocado_sn'
17:27:53 INFO | Running 'vgdisplay avocado_vg'
17:27:54 DEBUG| [stderr]   Volume group "avocado_vg" not found.
17:27:54 DEBUG| [stderr]   Cannot process volume group avocado_vg
17:27:54 INFO | Command 'vgdisplay avocado_vg' finished with 5 after 0.43906140327453613s
17:27:54 ERROR| Command 'vgdisplay avocado_vg' failed.
stdout: b''
stderr: b'  Volume group "avocado_vg" not found.\n  Cannot process volume group avocado_vg\n'
additional_info: None
17:27:54 INFO | Running 'lvdisplay avocado_vg'
17:27:54 DEBUG| [stderr]   Volume group "avocado_vg" not found.
17:27:54 DEBUG| [stderr]   Cannot process volume group avocado_vg
17:27:54 INFO | Command 'lvdisplay avocado_vg' finished with 5 after 0.4472520351409912s
17:27:54 INFO | Running 'lvdisplay avocado_vg'
17:27:55 DEBUG| [stderr]   Volume group "avocado_vg" not found.
17:27:55 DEBUG| [stderr]   Cannot process volume group avocado_vg
17:27:55 INFO | Command 'lvdisplay avocado_vg' finished with 5 after 0.44734883308410645s
17:27:55 INFO | Running 'losetup --find'
17:27:55 DEBUG| [stdout] /dev/loop6
17:27:55 INFO | Command 'losetup --find' finished with 0 after 0.0009458065032958984s
17:27:55 INFO | Running 'dd if=/dev/zero of=tmp_loop6.img bs=4096 count=262144'
17:27:55 DEBUG| [stderr] 262144+0 records in
17:27:55 INFO | Command 'dd if=/dev/zero of=tmp_loop6.img bs=4096 count=262144' finished with 0 after 0.541276216506958s
17:27:55 DEBUG| [stderr] 262144+0 records out
17:27:55 DEBUG| [stderr] 1073741824 bytes (1.1 GB, 1.0 GiB) copied, 0.540485 s, 2.0 GB/s
17:27:55 INFO | Running 'losetup /dev/loop6 tmp_loop6.img -P'
17:27:55 INFO | Command 'losetup /dev/loop6 tmp_loop6.img -P' finished with 0 after 0.04853677749633789s
17:27:55 DEBUG| PARAMS (key=lv_snapshot_size, path=*, default=460.8) => 460.8
17:27:55 INFO | Running 'vgdisplay avocado_vg'
17:27:56 DEBUG| [stderr]   Volume group "avocado_vg" not found.
17:27:56 DEBUG| [stderr]   Cannot process volume group avocado_vg
17:27:56 INFO | Command 'vgdisplay avocado_vg' finished with 5 after 1.128328561782837s
17:27:56 ERROR| Command 'vgdisplay avocado_vg' failed.
stdout: b''
stderr: b'  Volume group "avocado_vg" not found.\n  Cannot process volume group avocado_vg\n'
additional_info: None
17:27:56 INFO | Running 'vgcreate avocado_vg /dev/loop6'
17:27:57 DEBUG| [stdout]   Physical volume "/dev/loop6" successfully created.
17:27:57 DEBUG| [stdout]   Volume group "avocado_vg" successfully created
17:27:58 INFO | Command 'vgcreate avocado_vg /dev/loop6' finished with 0 after 1.247370958328247s
```

**Comment by narasimhan-v** [comment]
```
# avocado run lvsetup.py
JOB ID     : f9ffd7c1a78c02e99d4fb6f811c41128d88ed1ca
JOB LOG    : /root/sim/tests/results/job-2020-06-24T19.00-f9ffd7c/job.log
 (1/3) lvsetup.py:Lvsetup.test: PASS (26.13 s)
 (2/3) lvsetup.py:Lvsetup.test_vg_reactivate: PASS (69.31 s)
 (3/3) lvsetup.py:Lvsetup.test_lv_snapshot: PASS (34.06 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/sim/tests/results/job-2020-06-24T19.00-f9ffd7c/results.html
JOB TIME   : 132.61 s
```

```
19:00:22 DEBUG| PARAMS (key=lv_disks, path=*, default=None) => None
19:00:22 DEBUG| PARAMS (key=vg_name, path=*, default=avocado_vg) => 'avocado_vg'
19:00:22 DEBUG| PARAMS (key=lv_name, path=*, default=avocado_lv) => 'avocado_lv'
19:00:22 DEBUG| PARAMS (key=fs, path=*, default=ext4) => 'ext4'
19:00:22 DEBUG| PARAMS (key=lv_size, path=*, default=0) => '0'
19:00:22 DEBUG| PARAMS (key=lv_snapshot_name, path=*, default=avocado_sn) => 'avocado_sn'
19:00:22 INFO | Running 'vgdisplay avocado_vg'
19:00:23 DEBUG| [stderr]   Volume group "avocado_vg" not found.
19:00:23 DEBUG| [stderr]   Cannot process volume group avocado_vg
19:00:23 INFO | Command 'vgdisplay avocado_vg' finished with 5 after 0.6626222133636475s
19:00:23 ERROR| Command 'vgdisplay avocado_vg' failed.
stdout: b''
stderr: b'  Volume group "avocado_vg" not found.\n  Cannot process volume group avocado_vg\n'
additional_info: None
19:00:23 INFO | Running 'lvdisplay avocado_vg'
19:00:23 DEBUG| [stderr]   Volume group "avocado_vg" not found.
19:00:23 DEBUG| [stderr]   Cannot process volume group avocado_vg
19:00:23 INFO | Command 'lvdisplay avocado_vg' finished with 5 after 0.6271445751190186s
19:00:23 INFO | Running 'lvdisplay avocado_vg'
19:00:24 DEBUG| [stderr]   Volume group "avocado_vg" not found.
19:00:24 DEBUG| [stderr]   Cannot process volume group avocado_vg
19:00:24 INFO | Command 'lvdisplay avocado_vg' finished with 5 after 0.5471360683441162s
19:00:24 INFO | Running 'losetup --find'
19:00:24 DEBUG| [stdout] /dev/loop6
19:00:24 INFO | Command 'losetup --find' finished with 0 after 0.000942230224609375s
19:00:24 DEBUG| [stdout] dd if=/dev/zero of=./tmp_loop6.img bs=4096 count=262144
19:00:24 DEBUG| [stdout]

19:00:24 INFO | Running 'dd if=/dev/zero of=./tmp_loop6.img bs=4096 count=262144'
19:00:25 DEBUG| [stderr] 262144+0 records in
19:00:25 INFO | Command 'dd if=/dev/zero of=./tmp_loop6.img bs=4096 count=262144' finished with 0 after 0.5428049564361572s
19:00:25 DEBUG| [stderr] 262144+0 records out
19:00:25 DEBUG| [stderr] 1073741824 bytes (1.1 GB, 1.0 GiB) copied, 0.541989 s, 2.0 GB/s
19:00:25 INFO | Running 'losetup /dev/loop6 ./tmp_loop6.img -P'
19:00:25 INFO | Command 'losetup /dev/loop6 ./tmp_loop6.img -P' finished with 0 after 0.06651425361633301s
19:00:25 DEBUG| PARAMS (key=lv_snapshot_size, path=*, default=460.8) => 460.8
19:00:25 INFO | Running 'vgdisplay avocado_vg'
19:00:25 DEBUG| [stderr]   Volume group "avocado_vg" not found.
19:00:25 DEBUG| [stderr]   Cannot process volume group avocado_vg
19:00:26 INFO | Command 'vgdisplay avocado_vg' finished with 5 after 1.207308053970337s
19:00:26 ERROR| Command 'vgdisplay avocado_vg' failed.
stdout: b''
stderr: b'  Volume group "avocado_vg" not found.\n  Cannot process volume group avocado_vg\n'
additional_info: None
19:00:26 INFO | Running 'vgcreate avocado_vg /dev/loop6'
19:00:27 DEBUG| [stdout]   Physical volume "/dev/loop6" successfully created.
19:00:27 DEBUG| [stdout]   Volume group "avocado_vg" successfully created
19:00:27 INFO | Command 'vgcreate avocado_vg /dev/loop6' finished with 0 after 1.1575875282287598s
```

File: io/disk/lvsetup.py [hunk 1]
```diff
@@ -110,39 +111,22 @@ def setUp(self):
             self.device = self.disk
         else:
             if not self.lv_size:
-                self.lv_size = 1024
+                self.lv_size = 1024 * 1024 * 1024
```
Comment (Naresh-ibm) [inline io/disk/lvsetup.py:114]: @narasimhan-v any reason for using (1024 * 1024 * 1024) ?
I mean if we need the value of this mathematical operation then we can directly use the result of above multiplication.
if we cannot use the the result value and if it is mandatory to use like in above formate then can you please add a comment here quoting the same reason?

Comment (narasimhan-v) [reply io/disk/lvsetup.py:114]: @narasimhan-v It is easy to understand this (3 times 1024 means it is GB) than `1073741824`, IMO.
I think the multiplied value is what needs comment, and this is easily understandable.

**Comment by narasimhan-v** [comment]
utils merged

**Comment by narasimhan-v** [comment]
@harish-24 review please

**Comment by narasimhan-v** [comment]
@PraveenPenguin Travis has passed, but not sure why it is showing as incomplete. Can you help here please.

**Pull request merged**